osu.Game/OsuGameBase.cs

```diff
@@ -586,11 +586,16 @@ private void onRulesetChanged(ValueChangedEvent<RulesetInfo> r)
                 return;
             }
 
+            var previouslySelectedMods = SelectedMods.Value.ToArray();
+
             if (!SelectedMods.Disabled)
                 SelectedMods.Value = Array.Empty<Mod>();
 
             AvailableMods.Value = dict;
 
+            if (!SelectedMods.Disabled)
+                SelectedMods.Value = previouslySelectedMods.Select(m => instance.CreateModFromAcronym(m.Acronym)).Where(m => m != null).ToArray();
```

Comment:
This implementation would work wrong if you had two ruleset-specific mods that _aren't_ the same mod, but which _happen to share_ a mod acronym.

I don't believe that this is a problem _right now_, but it may become one later on. Not sure if this is a big deal, but I'm just saying that there's a big implicit assumption here that makes it work as-is.

Reply:
If two different mods share the same acronym, then that's a whole other problem of its own. An acronym should point out mods that share the same idea and principles, and shouldn't be different across different rulesets, but I can see how that could happen (with custom rulesets especially).

I could use the mod's name instead, but went with acronym since there's a specific factory method for it.

Reply:
I think we already have tests in place to avoid two rulesets using the same acronym for different mods (as it can lead to issues with serialisation when falling back to acronyms)?

Reply:
I think that one was about all mods of *each* ruleset not having the same acronym, while the concern here is about two rulesets having two mods meant to do two completely different things (i.e. have two different names) but share the same acronym. Maybe it would make sense to have a validation test that ensures that all mod acronyms map to the same name.

Reply:
I can only find [this](https://github.com/ppy/osu/blob/master/osu.Game.Tests/Visual/Gameplay/TestSceneModValidity.cs#L17-L24), which as mentioned above, only guarantees uniqueness of acronyms *within* the context of one ruleset, and not uniqueness among all possible rulesets.

> Maybe it would make sense to have a validation test that ensures that all mod acronyms map to the same name, if acronyms are still meant to do that.

I have no idea how you would write this, since technically, as things stand, `OsuModDoubleTime` and `TaikoModDoubleTime` would technically conflict.

Reply:
As in a test would go through all mods in all rulesets and take the name and acronym of them, and the ensure that all mods with the same acronym have the same name.

Reply:
I think it's fine to go with this assumption for now. This really improves the usability of mod selection overall.